File: src/main/java/apache/ignite/futures/topicmessage/Result.java
package apache.ignite.futures.topicmessage;

/** Result sent from the server to the client. */
class Result<V> {
    /** Result. */
    private final V value;

    /** Failure. */
    private final String failure;

    /** Constructor. */
    Result(V value, String failure) {
        this.value = value;
        this.failure = failure;
    }

    /** @return Result or {@code null} if the operation has no result or the operation failed. */
    V value() {
        return value;
    }

    /** @return Failure or {@code null} if the operation was successful. */
    String failure() {
        return failure;
    }
}

File: Apache.Ignite.Futures/TopicMessage/Result.cs
namespace Apache.Ignite.Futures.TopicMessage
{
    /// <summary>
    /// Result sent from the server to the client.
    /// </summary>
    public class Result
    {
        /// <summary>
        /// Gets or sets result or <code>null</code> if the operation has no result or the operation failed.
        /// </summary>
        public object Value { get; set; }

        /// <summary>
        /// Gets or sets failure or <code>null</code> if the operation was successful.
        /// </summary>
        public string Failure { get; set; }
    }
}
